### PR TITLE
Defined additional semantic convention attribute names

### LINKED ISSUE #362: Miscellaneous additional semantic convention attribute names
As I am working on various implementations I have found a number attribute names that are not currently specified or have an existing open issue.

**http.user_agent** - present in OpenCensus, absolutely required by systems my team is currently working on

**iam.user** - username value from basic authentication or subject from access/identity token, absolutely required by systems my team is currently working on

**iam.role** - scope/authorities from access token, not needed by my team but useful for completeness of the specification

**message_bus.destination** - present in OpenTracing, absolutely required by systems my team is currently working on

Also needed is more standardization of "component" attribute values. This already a issue to standardize on **db** for database spans. Other needs we have found are:

**messaging** - asynchronous messaging systems
**rpc** - instead of **grpc** or if grpc is kept then also needed is **rmi**, **thrift**, etc.
**cluster** - or maybe **streaming** for internal cross node communications on streaming platforms

### PR DESCRIPTION
**Pull request by kbrockhoff**
Resolves #362

### CODE REVIEW

**Comment by Oberon00** [comment]
I recommend splitting this PR up, e.g. the messaging changes probably deserve their own PR.

File: specification/data-events.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Semantic conventions for events
```
Comment (Oberon00) [inline specification/data-events.md]: I would put the events to the places they belong to, e.g. message events to event conventions, error to general conventions, etc.

We also have status, name and attributes already bundled together and grpc has events right with the other conventions too.

File: specification/data-events.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Most exporters will likely drop the `message.content` attribute if present.
+However, logging-only exporters will likely want to log it as this information
+is highly useful during the early stages of developing a new application.
+
+## Error event attributes
+
+Event `"name"` MUST be `"error"`.
+
+Error events MAY be associated with a tracing span or recorded independently.
```
Comment (tigrannajaryan) [inline specification/data-events.md]: How can events be recorded independently? I do not think OpenTelemetry currently has a concept of free-standing event. AFAIK this is something that is planned to be added, but does not exist yet.
The API specification currently defines Events that can be associated with Spans: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#add-events

File: specification/data-async-messaging.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Semantic conventions for asynchronous messaging spans
+
+For call publishing a message to the bus, the `SpanKind` MUST be `Producer`.
+For call retrieving a message from the bus, the `SpanKind` MUST be `Consumer`.
+
+## Common Attributes
+
+| Attribute name | Notes and examples                                           | Required? |
+| :------------- | :----------------------------------------------------------- | --------- |
+| `component`    | Denotes the type of the span and needs to be `"bus"`. | Yes |
```
Comment (tigrannajaryan) [inline specification/data-async-messaging.md]: There is a proposal to remove "component" attribute: https://github.com/open-telemetry/opentelemetry-specification/issues/336
Comments on that issue are welcome.

File: specification/data-events.md (hunk 2 above)
Comment (kbrockhoff) [reply specification/data-events.md]: Removed

File: specification/data-async-messaging.md (hunk 3 above)
Comment (kbrockhoff) [reply specification/data-async-messaging.md]: Removed

File: specification/data-events.md (hunk 1 above)
Comment (kbrockhoff) [reply specification/data-events.md]: Message events apply to all types of remote calls, not only gRPC.

**Comment by kbrockhoff** [comment]
I will split up this pull request into multiple requests.

**Pull request closed without merging**